src/StreamrClient.js

```diff
@@ -570,10 +570,12 @@ export default class StreamrClient extends EventEmitter {
                     // another resend if it doesn't. So we can anyway clear this resend request.
```

Comment:
will this work correctly now though? If messages haven't been stored yet, shouldn't the network say "resend done" and "no_resend"? Then we *should* do the secondResend, but also we should *not* swap HistoricalSubscription for RealtimeSubscription yet, I think? Only after second resend confirms there really was nothing to resend?

Reply:
I think you're right. The following scenario is valid:
1) subscribe with resend options
2) a first resend request is sent
3) the network answers with "no_resend" but maybe some messages are still being propagated and waiting to be stored
4) The HistoricalSubscription should NOT emit "resend done"
5) After some delay, the second resend request is sent
6) The network again answers with "no_resend". We assume the delay is large enough so that this response is correct.
7) This time the HistoricalSubscription can emit "resend done" and we switch to RealTimeSubscription

I added a test to reproduce your reported bug and the scenario above. I did a small fix in HistoricalSubscription to be able to differentiate between the first and the second "no_resend" answers from the network.